### PR TITLE
Add Thai2Rom ONNX model

### PR DESCRIPTION
**Pull request by new5558**
### What does this changes

Add ONNX model for Thai2ROM per #639 

image

Initial results looks great (2 times speed up)

### What was wrong

Description of what was the root cause of the issue.

### How this fixes it

Description of how the changes fix the issue.

Fixes #...

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](https://github.com/PyThaiNLP/pythainlp/blob/dev/CONTRIBUTING.md) to this repository.

- [ ] Passed code styles and structures
- [x] Passed code linting checks and unit test

### What I need do before this PR ready
- [x] Run full code coverage test
- [x] Run linting and formatter
- [x] Upload ONNX file to my temporary storage
- [x] Add scripts I used to convert to ONNX file
- [x] Convert pytorch code to numpy from ONNX class
- [x] Refactor code to another file?
- [x] Others? Please suggest
- [x] Add Documentation for ONNX Thai2ROM

### CODE REVIEW

**Comment by new5558** [comment]
Here are the ONNX files: https://drive.google.com/drive/folders/1e6KCFOPs2pxQAarkK1DrWmvch807jKRI?usp=sharing


Could the repo's maintainer upload these two files to PyThaiNLP cloud storage so I can link them to the code?

Edited: Or should I just commit the models (total 4Mbs) to git?

**Comment by wannaphong** [comment]
> Here are the ONNX files: https://drive.google.com/drive/folders/1e6KCFOPs2pxQAarkK1DrWmvch807jKRI?usp=sharing
> 
> Could the repo's maintainer upload these two files to PyThaiNLP cloud storage so I can link them to the code?
> 
> Edited: Or should I just commit the models (total 4Mbs) to git?


OK 👍  You can push file to git.

File: pythainlp/transliterate/thai2rom.py [hunk 1]
```diff
@@ -3,9 +3,6 @@
 Romanization of Thai words based on machine-learnt engine ("thai2rom")
 """
 
-import random
-
-import numpy as np
```
Comment (new5558) [inline pythainlp/transliterate/thai2rom.py:8]: Original model need to be removed `numpy` from `forward()` for ONNX conversion to work properly

File: pythainlp/transliterate/thai2rom_onnx.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+"""
+Romanization of Thai words based on machine-learnt engine in ONNX runtime ("thai2rom")
+"""
+from pythainlp.corpus import get_corpus_path
+import numpy as np
+import json
+
+from onnxruntime import InferenceSession
+
+_MODEL_ENCODER_NAME = "thai2rom_encoder_onnx"
```
Comment (new5558) [inline pythainlp/transliterate/thai2rom_onnx.py:11]: My Initial attempt to convert the whole model to ONNX failed (convert success but model prediction result not matched with original model). So I seperately convert encoder and decoder.

File: pythainlp/transliterate/thai2rom_onnx.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        Transliteration of Thai words.
+
+        Now supports Thai to Latin (romanization)
+        """
+        # get the model, will download if it's not available locally
+        self.__encoder_filename = get_corpus_path(_MODEL_ENCODER_NAME)
+        self.__decoder_filename = get_corpus_path(_MODEL_DECODER_NAME)
+        self.__config_filename = get_corpus_path(_MODEL_CONFIG_NAME)
+
+        print(get_corpus_path(_MODEL_DECODER_NAME))
```
Comment (wannaphong) [inline pythainlp/transliterate/thai2rom_onnx.py]: Can you close this line?

Comment (new5558) [reply pythainlp/transliterate/thai2rom_onnx.py]: Sorry, Done krub.

**Review by wannaphong — APPROVED** [review]
Awesome 👍

**Pull request merged**